### PR TITLE
[bug] Allow `__opencl_c_named_address_space_builtins' feature

### PR DESCRIPTION
**Pull request by zhoujingya**
#26

### CODE REVIEW

File: lib/CL/devices/pthread/pthread.c [hunk 1]
```diff
@@ -71,7 +71,6 @@ struct data {
 void
 pocl_pthread_init_device_ops(struct pocl_device_ops *ops)
 {
-  pocl_ventus_init_device_ops(ops);
```
Comment (Jules-Kong) [inline lib/CL/devices/pthread/pthread.c:74]: Removed redundant code to prevent pocl_ventus_init from being called twice. Please put a separate commit.

File: lib/CL/devices/ventus/pocl_ventus.cc [hunk 2]
```diff
@@ -249,11 +249,13 @@ pocl_ventus_init (unsigned j, cl_device_id dev, const char* parameters)
   dev->linker_available = CL_TRUE;
 
   char extensions[1024];
+  char features[1024];
```
Comment (Jules-Kong) [inline lib/CL/devices/ventus/pocl_ventus.cc:252]: please add features[0]=0;

File: lib/CL/devices/ventus/pocl_ventus.cc [hunk 3]
```diff
@@ -249,11 +249,14 @@ pocl_ventus_init (unsigned j, cl_device_id dev, const char* parameters)
...
-  strcat (extensions, "cl_khr_fp64"
-                      " __opencl_c_generic_address_space");
-  dev->extensions = strdup (extensions); // no extention support now
-
+  features[0] = 0;
+  strcat (extensions, "cl_khr_fp64 ");
+  strcat(features, "__opencl_c_generic_address_space "
+                   " __opencl_c_named_address_space_builtins");
+  dev->extensions = strdup (extensions);
+  dev->features = strdup(features);
```
Comment (Jules-Kong) [inline lib/CL/devices/ventus/pocl_ventus.cc:259]: Maybe you can try to delete this part of the code modification and keep only the 1156 lines of code to see if the effect is equivalent.

File: lib/CL/devices/pthread/pthread.c (hunk 1 above)
Comment (Jules-Kong) [reply lib/CL/devices/pthread/pthread.c]: “extensions" is recommended.

File: lib/CL/devices/ventus/pocl_ventus.cc [hunk 4]
```diff
@@ -249,11 +249,14 @@ pocl_ventus_init (unsigned j, cl_device_id dev, const char* parameters)
   dev->linker_available = CL_TRUE;
 
   char extensions[1024];
+  char features[1024];
   extensions[0] = 0;
-  strcat (extensions, "cl_khr_fp64"
-                      " __opencl_c_generic_address_space");
-  dev->extensions = strdup (extensions); // no extention support now
-
```
Comment (Jules-Kong) [inline lib/CL/devices/ventus/pocl_ventus.cc:256]: “extensions" is recommended.

File: lib/CL/devices/ventus/pocl_ventus.cc [hunk 5]
```diff
@@ -249,11 +249,14 @@ pocl_ventus_init (unsigned j, cl_device_id dev, const char* parameters)
...
+  features[0] = 0;
+  strcat (extensions, "cl_khr_fp64 ");
+  strcat(features, "__opencl_c_generic_address_space "
```
Comment (Jules-Kong) [inline lib/CL/devices/ventus/pocl_ventus.cc:256]: The format remains consistent.

File: lib/CL/pocl_llvm_build.cc [hunk 6]
```diff
@@ -368,6 +368,8 @@ int pocl_llvm_build_program(cl_program program,
     exts += ' ';
     exts += device->features;
   }
+  if (strcmp(device->llvm_cpu, "ventus-gpgpu") == 0)
```
Comment (Jules-Kong) [inline lib/CL/pocl_llvm_build.cc:371]: Recommended to delete.

**Review by yangzexia — APPROVED** [review]
LGTM

**Comment by zhoujingya** [comment]
> LGTM

小孔说。这个`pocl_llvm_build.cc`文件里的这个函数`pocl_llvm_build_program`里面其实是已经为编译器构建了编译参数的，只是我们目前没有走这套逻辑，所以需要自己加，这个PR先合进去，因为不太确定啥时候改目前pocl的build_program的实现

**Comment by Jules-Kong** [comment]
https://github.com/THU-DSP-LAB/pocl/pull/52
